tests/chart_tests/test_global_storageclass.py

```diff
@@ -28,5 +28,23 @@ def test_global_storageclass(supported_types, expected_output):
 
     assert all(
         expected_output in storageClassNames
-        for storageClassNames in jmespath.search("[*].spec.volumeClaimTemplates[*].spec.storageClassName", docs)
+        for storageClassNames in jmespath.search("[].spec.volumeClaimTemplates[].spec.storageClassName", docs)
```

Comment:
Ugh. jmespath is so unreliable in python that any time we modify it, I can't trust that what I understand what's happening.

Reply:
I think we should rework this test to not use jmespath, and instead use comprehensions or some other way to extract data where we know what we're getting. We can work on this together if you want. I have a datastructure in mind that will make this a lot easier to understand and be confident with, which is like:

```
input_data = (
    ("charts/alertmanager/templates/alertmanager-statefulset.yaml", "astrosc"),
    ("charts/elasticsearch/templates/master/es-master-statefulset.yaml", "astrosc"),
    ("charts/elasticsearch/templates/data/es-data-statefulset.yaml", "astrosc"),
    ("charts/prometheus/templates/prometheus-statefulset.yaml", "astrosc"),
    ("charts/astronomer/templates/registry/registry-statefulset.yaml", None),
)
```

Where the parameters are "show_only" and "expected_sc_name".

Also that "expected_types" var was probably copied and pasted from somewhere else and needs to be changed to something that makes sense here.